stubs/urllib3/urllib3/connectionpool.pyi

```diff
@@ -29,48 +34,52 @@ Retry = retry.Retry
 Timeout = timeout.Timeout
 get_host = url.get_host
 
+_Timeout: TypeAlias = Timeout | float
+_Retries: TypeAlias = Retry | bool | int
+
 xrange: Any
-log: Any
+log: Logger
 
 class ConnectionPool:
-    scheme: Any
-    QueueCls: Any
-    host: Any
-    port: Any
-    def __init__(self, host, port=...) -> None: ...
-    def __enter__(self): ...
-    def __exit__(self, exc_type, exc_val, exc_tb): ...
-    def close(self): ...
+    scheme: ClassVar[str | None]
+    QueueCls: ClassVar[Any]
+    host: str
+    port: int | None
+    def __init__(self, host: str, port: int | None = ...) -> None: ...
+    def __enter__(self: Self) -> Self: ...
+    def __exit__(self, exc_type, exc_val, exc_tb) -> bool: ...
+    def close(self) -> None: ...
 
 class HTTPConnectionPool(ConnectionPool, RequestMethods):
-    scheme: Any
-    ConnectionCls: Any
-    strict: Any
-    timeout: Any
-    retries: Any
-    pool: Any
-    block: Any
-    proxy: Any
-    proxy_headers: Any
-    num_connections: Any
-    num_requests: Any
+    scheme: ClassVar[str]
+    ConnectionCls: ClassVar[type[HTTPConnection]]
```

Comment:
I believe the `Union` type on the main branch is deliberate here: while `HTTPSConnectionPool`s can only have `HTTPSConnection`s, `HTTPConnectionPool`s can have `HTTPConnection`s or `HTTPSConnection`s: https://github.com/urllib3/urllib3/blob/1e985050bb498f15895478bd61acc002be83ae65/src/urllib3/connectionpool.py#L169

```suggestion
    ConnectionCls: ClassVar[type[HTTPConnection | HTTPSConnection]]
```